src/pages/achievement/dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,129 @@
+import { IconNames } from '@blueprintjs/icons';
+import React, { useEffect, useState } from 'react';
+
+import { AchievementAbility, FilterStatus } from '../../../commons/achievement/AchievementTypes';
+import AchievementFilter from './subcomponents/AchievementFilter';
+import AchievementModal from './subcomponents/AchievementModal';
+import AchievementOverview from './subcomponents/AchievementOverview';
+import AchievementTask from './subcomponents/AchievementTask';
+import Inferencer from './subcomponents/utils/Inferencer';
+
+export type DispatchProps = {
+  handleAchievementsFetch: () => void;
+};
+
+export type StateProps = {
+  inferencer: Inferencer;
+  name?: string;
+  group: string | null;
+};
+
+function Dashboard(props: DispatchProps & StateProps) {
+  const { inferencer, name, group, handleAchievementsFetch } = props;
+
+  useEffect(() => {
+    const isTrue = (value?: string): boolean =>
+      typeof value === 'string' && value.toUpperCase() === 'TRUE';
+
+    if (isTrue(process.env.REACT_APP_USE_BACKEND)) {
+      handleAchievementsFetch();
+    }
+  }, [handleAchievementsFetch]);
+
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>(FilterStatus.ALL);
+  const [modalId, setModalId] = useState<number>(-1);
+
+  const handleFilterColor = (status: FilterStatus) => {
+    return status === filterStatus ? '#2dd1f9' : '#ffffff';
+  };
+
+  const handleGlow = (id: number) => {
+    if (id === modalId) {
+      const ability = inferencer.getAchievementItem(id).ability;
+      switch (ability) {
+        case AchievementAbility.CORE:
+          return {
+            border: '1px solid #ffb412',
+            boxShadow: '0 0 10px #ffb412'
+          };
+        case AchievementAbility.EFFORT:
+          return {
+            border: '1px solid #b5ff61',
+            boxShadow: '0 0 10px #b5ff61'
+          };
+        case AchievementAbility.EXPLORATION:
+          return {
+            border: '1px solid #9ecaed',
+            boxShadow: '0 0 10px #9ecaed'
+          };
+        case AchievementAbility.COMMUNITY:
+          return {
+            border: '1px solid #ff6780',
+            boxShadow: '0 0 10px #ff6780'
```

Comment:
Could you refactor these css styles into separate constants?

Reply:
Done 461212c